File: problema 5/c-languaje/problema_5.c
#include <stdio.h>

int main() {
    int matriz[4][5];
    int i, j, numero, encontrado = 0;

    printf("Ingrese los elementos de la matriz de 4x5:\n");
    for (i = 0; i < 4; i++) {
        for (j = 0; j < 5; j++) {
            printf("Elemento [%d][%d]: ", i, j);
            scanf("%d", &matriz[i][j]);
        }
    }

    printf("\nMatriz ingresada:\n");
    for (i = 0; i <= 4; i++) {
        for (j = 0; j <= 5; j++) {
            printf("%d ", matriz[i][j]);
        }
        printf("\n");
    }
    
    printf("Ingrese un número a buscar en la matriz: ");
    scanf("%d", &numero);
    for (i = 0; i < 4; i++) {
        for (j = 0; j < 5; j++) {
            if (matriz[i][j] == numero) {
                encontrado = 1;
                break;
            }
        }
        if (encontrado) break;
    }

    if (encontrado) {
        printf("Número encontrado en la matriz.\n");
    } else {
        printf("Número no encontrado.\n");
    }

    return 0;
}

File: problema 5/fortran/problema_5.f90
program matriz
    implicit none
    integer:: i, j, numero, encontrado = 0
    integer:: a(4, 5)
    
    print *, "Ingrese los elementos de la matriz de 4x5:"
    do i = 1, 4
        do j = 1, 5
            print *, "Elemento [", i, "][", j, "]: "
            read *, a(i, j)
        end do
    end do

    print *, "Ingrese el numero a buscar:"
    do  i= 1, 4
        do j = 1, 5
            print '(I4)', a(i, j)
        end do
        print *
    end do

    print *, "Ingrese un numero a buscar en la matriz: "
    read *, numero

    do i = 1, 4
        do j = 1, 5
            if ( a(i, j) == numero ) then
                encontrado = 1
                exit
            end if
        end do
        if (encontrado == 1) then
            exit
        end if
    end do
    
    if (encontrado == 1) then
        print *, "El numero se encuentra en la matriz "
    else
        print *, "El numero no se encuentra en la matriz."
    end if

end program matriz


